internal/hcsoci/layers.go

```diff
@@ -127,10 +152,12 @@ func MountContainerLayers(ctx context.Context, layerFolders []string, guestRoot
 
 	hostPath := filepath.Join(layerFolders[len(layerFolders)-1], "sandbox.vhdx")
 	containerScratchPathInUVM := ospath.Join(uvm.OS(), guestRoot)
-	_, _, containerScratchPathInUVM, err = uvm.AddSCSI(ctx, hostPath, containerScratchPathInUVM, false)
+	scsiMount, err := uvm.AddSCSI(ctx, hostPath, containerScratchPathInUVM, false)
```

Comment:
I think we need to do more refactoring on the container layers code. We are currently just throwing away the resource types we get from `AddVSMB` and `AddSCSI` in here. We should be saving these and returning them either directly or nested as members of another resource type.

Reply:
It also looks like `AddSCSILayer` should be changed to return a `SCSIMount`.

Reply:
Could be nested in ImageLayers

Reply:
This work could be grouped in with the unexporting of the Remove<Resource> functions. https://github.com/microsoft/hcsshim/pull/772#discussion_r390750000 Would probably be better to do it all in another pr.